### PR TITLE
build: update to Angular 15 rc

### LINKED ISSUE #3614: Build: Update to latest Angular v15 RC
### Which @ngrx/* package(s) are relevant/related to the feature request?

component-store, component, data, effects, entity, router-store, schematics, store-devtools, store

### Information

Update necessary dependencies manually using ng update through Nx.

```sh
FORCE_NG_UPDATE=true ng update @angular/cli @angular/core --next
```

### Describe any alternatives/workarounds you're currently using

_No response_

### I would be willing to submit a PR to fix this issue

- [ ] Yes
- [ ] No

### PR DESCRIPTION
**Pull request by markostanimirovic**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/ngrx/platform/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Documentation has been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[x] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

Closes #3614 

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: .node-version [hunk 1]
```diff
@@ -1 +1 @@
-14.17
\ No newline at end of file
+14.20
```
Comment (markostanimirovic) [inline .node-version:1]: Angular v15 suports Node v14.20 or higher. I also updated `"engines"` in `package.json` to be the same as in the Angular repo.

File: modules/effects/src/provide_effects.ts [hunk 2]
```diff
@@ -41,35 +46,33 @@ import { rootEffectsInit as effectsInit } from './effects_actions';
...
-  return {
-    ɵproviders: [
-      effects,
-      {
-        provide: ENVIRONMENT_INITIALIZER,
-        multi: true,
-        useValue: () => {
-          inject(ROOT_STORE_PROVIDER);
-          inject(FEATURE_STATE_PROVIDER, { optional: true });
+  return makeEnvironmentProviders([
```
Comment (markostanimirovic) [inline modules/effects/src/provide_effects.ts:49]: Angular v15 now exports the `EnvironmentProviders` type, also there is the `makeEnvironmentProviders` function that converts providers array to environment providers.

File: modules/router-store/src/serializers/full_serializer.ts [hunk 3]
```diff
@@ -26,6 +26,7 @@ export class FullRouterStateSerializer
       data: route.data,
       url: route.url,
       outlet: route.outlet,
+      title: route.title,
```
Comment (markostanimirovic) [inline modules/router-store/src/serializers/full_serializer.ts:29]: In v15, `title` is required property

File: modules/schematics/src/container/index.ts [hunk 4]
```diff
@@ -83,21 +81,12 @@ function addStateToComponent(options: Partial<ContainerOptions>) {
...
-    const storeConstructor = [start, `(private store: Store)`, end].join('');
+    ) as ts.ClassDeclaration;
     const constructorUpdate = new ReplaceChange(
       componentPath,
-      pos,
-      `  ${constructorText}\n\n`,
-      `\n\n  ${storeConstructor}`
+      componentClass.members.pos,
+      '\n',
+      `\n  constructor(private store: Store) {}`
```
Comment (markostanimirovic) [inline modules/schematics/src/container/index.ts:89]: In v15, the component schematic doesn't generate the constructor and `ngOnInit` method anymore. Instead, it generates an empty component class.

**Review by brandonroberts — APPROVED** [review]
🚀

File: modules/router-store/src/serializers/full_serializer.ts (hunk 3 above)
Comment (brandonroberts) [reply modules/router-store/src/serializers/full_serializer.ts:29]: 🤔

Comment (markostanimirovic) [reply modules/router-store/src/serializers/full_serializer.ts:29]: *`title` is required property in `ActivatedRouteSnapshot` class, not `Route` interface 😅

Comment (brandonroberts) [reply modules/router-store/src/serializers/full_serializer.ts:29]: Ahh ok. I was worried there for a minute 😄

**Review by timdeschryver — APPROVED** [review]

**Pull request merged**